### ISSUE TITLE
firebase_auth: Attempt to invoke interface method 'long com.google.firebase.auth.FirebaseUserMetadata.getCreationTimestamp()' on a null object reference

### ISSUE THREAD

**Issue by KammererTob** [flutter/flutter#27246]
Some users complained about not being able to log in any more. I was not able to find any problems, but was able correlate one unsuccessful attempt to the exception below.
This most likely occurs when i try to call `FirebaseAuth.instance.currentUser()`

My crashlytics has recorded this issue 54 times for 9 different users, so assume that it isn't a fluke. 
It also seems to be only happening on Android, because i have no iOS mentioning that yet.
Is there something i can do?

## Steps to Reproduce

Sadly i cannot provide steps to reproduce because it only happens in production, only for a subset of users.

## Logs

```
PlatformException(error, Attempt to invoke interface method 'long com.google.firebase.auth.FirebaseUserMetadata.getCreationTimestamp()' on a null object reference, null)
```

```
[√] Flutter (Channel beta, v1.0.0, on Microsoft Windows [Version 10.0.17134.556], locale de-DE)
    • Flutter version 1.0.0 at C:\SDKs\flutter
    • Framework revision 5391447fae (9 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\SDKs\android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = C:\SDKs\android
    • Java binary at: D:\Programme\JDK\bin\java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_92-b14)
    • All Android licenses accepted.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[!] IntelliJ IDEA Ultimate Edition (version 2018.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2018.2.5
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[√] VS Code (version 1.30.2)
    • VS Code at C:\Users\puelo\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 2.22.1

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.0.0 (API 26) (emulator)

! Doctor found issues in 2 categories.
```

**Comment by zoechi** [comment]
When did the errors start?

**Comment by KammererTob** [comment]
I have only added crashlytics after users started having problems. The first review where it was mentioned dates back to 22nd of December. So i assume sometime around there

**Comment by zoechi** [comment]
Perhaps https://stackoverflow.com/questions/48079683/firebase-user-returns-null-metadata-for-already-signed-up-users

**Comment by KammererTob** [comment]
It might be that. Sadly i don't have any influence on how/if getMetadata() is called, because it is used here:
https://github.com/flutter/plugins/blob/master/packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java#L627

**Comment by KammererTob** [comment]
I am a little hesitant touching the gradle depency versions because of: https://github.com/flutter/flutter/issues/24708

And my dependencies are (apart from the gradle version 3.2.1 < 3.3.0) identical to those used in the example provided by the firebase_auth project. 
Also: Shouldn't a dependency problem affect all users? This bug somehow only affects a small subset of users.

**Comment by KammererTob** [comment]
I was able to somewhat confirm that it depends on the device used (not the actual accounts).
Thus a user was unable to login with a Smartphone, but it worked perfectly fine with a tablet. 
Sadly i was still unable to get my hands on a non-working device and thus the problem persists.

Any suggestions?

**Comment by KammererTob** [comment]
Just wanted to confirm that using a dependency override for firebase_auth, referencing the pull-request, solved the issues for our users.

**Comment by dotdoom** [comment]
That PR has been closed because it was a subset of https://github.com/flutter/plugins/pull/963 which is merged now.

**Comment by KammererTob** [comment]
Closed because merge of flutter/plugins#963 was released

**Comment by kroikie** [comment]
@KammererTob

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/478. Any further collaboration will be done there.
